### ISSUE TITLE
replace snowsql download with direct upload to S3

### ISSUE THREAD

**Issue by ErikZigo** [keboola/db-extractor-snowflake#1]
Ted to mam udelany tak, ze pro ziskani souboru pouziva extractor `snowsql cli`.

Pokud ma extractor vyuzivat rovnou export do S3 na urovni snowflaku, melo by se stat:

- [ ] Uprava KBC - param. platnost klicu pro upload souboru do S3 - https://github.com/keboola/connection/issues/721

- [ ] Uprava Docker Runeru - Novy parametr `output` pro `staging_storage`
Obdoba https://github.com/keboola/docker-bundle/pull/79 s tim, ze vytvorene AWS S3 klice budou mit platnost podle nasstaveneho timeoutu komponenty

- [ ] Uprava Extractoru - vyhodit `snowsql` a zaridit direct upload do S3
Vykrast https://github.com/keboola/transformation-bundle/blob/master/library/Keboola/Transformation/Backend/Snowflake.php#L527

- [ ] Rozsireni developer portalu  - Novy parametr `stagingStorageOutput`

- [ ] Pridat EX do AWS ECR

No a nakonec udelat
- [ ] UI pro extractor
Pripravit UI tak, aby automaticky udaje z jednotlivych queries ukladal do output mappingu

**Comment by Halama** [comment]
a format konfigurace extraktoru je připraven tak aby byl kompatibilní s output mappingem?

**Comment by ErikZigo** [comment]
Jak to myslis kompatibilni s output mappingem?

Vcera nez to vsechno spadlo se mi nepovedlo najit nikde dokumentaci k tomu `staging_storage` `s3`. Nevis kde to je?

**Comment by Halama** [comment]
No tak aby to fungovalo tak to musi byt tak ze to bude mit output mapping popsany takhle https://developers.keboola.com/extend/common-interface/config-file/#output-mapping-----basic

Ta nova feature dockeru runneru ti pak pro kazdou tam specifikovanout tabulku pripravi S3 credentials. Takova je zakladni idea, a je to potřeba pořádně popsat v issue u docker runneru.

**Comment by Halama** [comment]
hmm, dokumentaci staging_storage pro input taky nemuzu najit.

**Comment by Halama** [comment]
https://github.com/keboola/docker-bundle/issues/143

**Comment by ErikZigo** [comment]
Me prijde, ze to je pomerne dost veci.
Ja bych se spis priklanel k variante, ze se dojede ten extractor s jejich CLI. To mam lokalne otestovany, ze by snad melo fungovat. Chybi jen dodelat testy na travisu.

A tyhle vsechny veci se udelaj jako dalsi krok?

**Comment by Halama** [comment]
no a bude kompatibilni ta konfigurace?

**Comment by ErikZigo** [comment]
Tak se to zmigruje?
Kdyz to ted primarne potrebuje Martin Lepka, tak to nemusi byt public? Na tom se to odzkousi.

**Comment by Halama** [comment]
jo no tak ok. opet trochu zbytecna prace kdyz to stacilo rozmyslet na zacatku ale ted uz asi nic jineho nezbyva.

**Comment by ErikZigo** [comment]
Jsem cekal, ze to bude skoro stejne jednoduchy jako ostatni extractory.
Fakt jsem netusil, ze tam bude takovyhle zadrb, kdy bude potreba upravit dalsich pet mist.

Vy jste si mozna neco na zacatku pred mesicem rekli, ale ke me to neprobublalo. Realne jsem se k tomu dostal az v pondeli a zacal testama. Kdybych zacal po svem z druhe strany, vedel bych to o den driv.

**Comment by ondrejhlavacek** [comment]
Nic jsme si neříkali. To, že by se pro to měl dodělat ten staging, by mě asi taky z voleje nenapadlo. 
Konfigurace imho půjde udělat tak, že to bude 100% kompatibilní. Hodím na to dneska aspoň myšlenku...

**Comment by Halama** [comment]
jo ta konfigurace urcite pujde udelat kompatiblini.

**Comment by odinuv** [comment]
Pokud bude komptabilni konfigurace, a bude to pres CLI, tak to bude super, protoze ten docker-runner output mapping udelame poradne https://github.com/keboola/docker-bundle/issues/144

**Comment by pivnicek** [comment]
@odinuv I see in the docs that exchanging data via s3 is only available via input mapping.  If I understand the issue here correctly, for this we would need exchanging data via s3 in output mapping, right?

![image](https://user-images.githubusercontent.com/569139/63026224-12773c80-beab-11e9-8e06-b98722bf6f30.png)

the idea being that we'd write direct to s3 with the extractor, and then put in the output mapping something like
```
{
    "storage": {
        "output": {
            "tables": [
                {
                    "source": "out.c-main.data.csv",
                    "destination": "out.c-main.data"
                }
            ]
        }
    }
}
```

**Comment by odinuv** [comment]
yes, and that is currently not possible.

**Comment by pivnicek** [comment]
ok, how do we make it possible? :-).   I guess as a workaround for now the extractor can write to the s3, then download the file and output it normally.

**Comment by odinuv** [comment]
yes, i assumed that was the plan

**Comment by pivnicek** [comment]
is it a good idea to have a dedicated s3 iam user for a component?

**Comment by ondrejhlavacek** [comment]
I don't think so. Different projects may accidentally read/write data from each other.
